src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocContentLocationCheck.java

```diff
@@ -0,0 +1,240 @@
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties" file.
+     */
+    public static final String MSG_JAVADOC_CONTENT_FIRST_LINE = "javadoc.content.first.line";
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties" file.
+     */
+    public static final String MSG_JAVADOC_CONTENT_SECOND_LINE = "javadoc.content.second.line";
+
+    /**
+     * Specify the policy on placement of the Javadoc content.
+     */
+    private JavadocContentLocation location = JavadocContentLocation.SECOND_LINE;
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.BLOCK_COMMENT_BEGIN,
+        };
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public boolean isCommentNodesRequired() {
+        return true;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (isMultilineComment(ast) && JavadocUtil.isJavadocComment(ast)) {
+            final String commentContent = JavadocUtil.getJavadocCommentContent(ast);
+            final int blankLines = countBlankLines(commentContent);
+            if (blankLines >= 0) {
+                if (location == JavadocContentLocation.FIRST_LINE) {
+                    if (blankLines != 0) {
+                        log(ast, MSG_JAVADOC_CONTENT_FIRST_LINE);
+                    }
+                }
+                else if (blankLines != 1) {
+                    log(ast, MSG_JAVADOC_CONTENT_SECOND_LINE);
+                }
+            }
+        }
+    }
+
+    /**
+     * Setter to specify the policy on placement of the Javadoc content.
+     *
+     * @param value string to decode location from
+     * @throws IllegalArgumentException if unable to decode
+     */
+    public void setLocation(String value) {
+        location = JavadocContentLocation.valueOf(value.trim().toUpperCase(Locale.ENGLISH));
+    }
+
+    /**
+     * Checks if a DetailAST of type {@code TokenTypes.BLOCK_COMMENT_BEGIN} span
+     * more than one line. The node always has at least one child of the type
+     * {@code TokenTypes.BLOCK_COMMENT_END}.
+     *
+     * @param node node to check
+     * @return {@code true} for multi-line comment nodes
+     */
+    private static boolean isMultilineComment(DetailAST node) {
+        return node.getLineNo() != node.getLastChild().getLineNo();
+    }
+
+    /**
+     * Returns the index of the first non-blank line.
+     * All lines consists only of asterisks and whitespaces are treated as blank.
+     *
+     * @param commentContent Javadoc content to process
+     * @return the number of blank lines or {@code -1} if the comment is completely blank
+     */
+    private static int countBlankLines(String commentContent) {
+        int lineNo = 0;
+        int returnValue = -1;
```

Comment:
You do not need two variables for calculation of result, please use either of them , but only one.

Reply:
done

Reply:
I still see old code, with two variables

Reply:
Please use only one variable.

Reply:
@romani here we need two variables: one to count the number of rows, the other to mark the case when there is no content.

Reply:
code was refactored, looks ok, I have no ideas on howto make it better.